interaction-manager/src/lib_components/spatial-navigation.js

```diff
@@ -0,0 +1,325 @@
+                    break;
+                case 'right':
+                    newDistance = Math.hypot(el.x, el.y - focusedElement.y);
+                    oldDistance = Math.hypot(acc.x, acc.y - focusedElement.y);
+                    break;
+                case 'left':
+                    newDistance = Math.hypot(window.innerWidth - el.x, el.y - focusedElement.y);
+                    oldDistance = Math.hypot(window.innerWidth - acc.x, acc.y - focusedElement.y);
+                    break;
+            }
+            acc = newDistance < oldDistance ? el : acc;
+
+            return acc;
+        });
+    }
+
+    /**
+     * Moves the focus in the desired direction
+     * @param {string} direction
+     * @returns {void}
+     */
+    moveFocus(direction) {
+        if (!this.enabled) return;
+
+        const focusableGroup = this.getFocusableGroup();
+        const { x, y } = document.activeElement.getBoundingClientRect();
+
+        if (focusableGroup.length === 0) return;
+
+        let nextFocusableElement = focusableGroup.reduce((acc, el) => {
+            const deltaX = el.x - x;
+            const deltaY = el.y - y;
+            const angle = toDeg(Math.atan2(deltaY, deltaX));
+
+            if (this.getDirectionAngle(direction, angle)) {
+                if (!acc) acc = el;
+
+                const newDistance = Math.hypot(deltaX, deltaY);
+                const oldDistance = Math.hypot(acc.y - y, acc.x - x);
+                acc = newDistance < oldDistance ? el : acc;
+            }
+
+            return acc;
+        }, null);
+
+        if (!nextFocusableElement) nextFocusableElement = this.getClosestToEdge(direction, focusableGroup, { x, y });
+
+        nextFocusableElement.element.focus();
+    }
+
+    /**
+     * Get the angle range for the direction
+     * @param {string} direction
+     * @param {number} angle
+     * @returns {boolean}
+     */
+    getDirectionAngle(direction, angle) {
+        switch (direction) {
+            case 'down':
+                return angle > 0 && angle < 180;
+            case 'up':
+                return angle > -180 && angle < 0;
+            case 'left':
+                return angle < -90 || angle > 90;
+            case 'right':
+                return angle > -90 && angle < 90;
+        }
+    }
+
+    /**
+     * Registers actions and adds them to the keyboard and gamepad objects
+     */
+    registerKeyActions() {
+        directions.forEach((direction) => {
+            const callback = () => {
+                this.moveFocus(direction);
+            };
+            actions.register(`move-focus-${direction}`, callback);
+
+            keyboard.on({
+                keys: [`arrow_${direction}`],
+                callback: `move-focus-${direction}`,
+                type: 'press',
+            });
+
+            gamepad.on({
+                actions: [`playstation.d-pad-${direction}`],

```

Comment:
I see in **gamepad.js** that "xbox" gamepad mappings integration is intended also. Is this for another time maybe? Although in this case, the value of playstation/xbox**.d-pad-${direction}** is 1:1.

Is this lib intented to be extended to the needs of the "user"? The question is outside the scope of the comment above.

Reply:
The values of corresponding xbox and playstation keys will always be the same. The idea behind this is that it's an alias for easier use. For example if we press the X button on the playstation, this corresponds to the A button on the xbox and to the 0 index button. Then the user can decide what will be easier for them to use. 

A good idea will be to allow users to set their own aliases. Will think about that later

As for why I'm using the playstation alias in the spatial navigation file, there is no particular reason. I only used it because I needed to have a string with the direction to loop through.